crates/notion-core/src/distro/yarn.rs

```diff
@@ -132,22 +124,27 @@ impl Distro for YarnDistro {
             return Ok(Fetched::Already(self.version));
         }
 
-        let temp = tempdir_in(path::tmp_dir()?).unknown()?;
+        let tmp_root = path::tmp_dir()?;
+        let temp = tempdir_in(&tmp_root).with_context(|_| ErrorDetails::CreateTempDirError {
+            in_dir: tmp_root.to_string_lossy().to_string(),
+        })?;
         let bar = progress_bar(
             self.archive.origin(),
             &format!("v{}", self.version),
             self.archive
                 .uncompressed_size()
                 .unwrap_or(self.archive.compressed_size()),
         );
+        let version_string = self.version.to_string();
 
         self.archive
             .unpack(temp.path(), &mut |_, read| {
                 bar.inc(read as u64);
             })
-            .unknown()?;
-
-        let version_string = self.version.to_string();
+            .with_context(|_| ErrorDetails::UnpackArchiveError {
+                tool: String::from("Yarn"),
+                version: version_string.clone(),
```

Comment:
For errors that's tricky, because the error is returned all the way to the root of the executable (`main.rs` and `shim.rs`), but the objects that generated them are necessarily created higher in the stack, leading to lifetime issues. Worth looking into, however.

Reply:
Makes sense, I figured that might be the case.